clients/drcachesim/tools/filter/record_filter.h

```diff
@@ -248,11 +249,13 @@ class record_filter_t : public record_analysis_tool_t {
     inline uint64_t
     add_to_filetype(uint64_t filetype)
     {
-        if (stop_timestamp_ != 0) {
+        if (stop_timestamp_ != 0)
             filetype |= OFFLINE_FILE_TYPE_BIMODAL_FILTERED_WARMUP;
-        }
-        if (shard_type_ == SHARD_BY_CORE) {
+        if (shard_type_ == SHARD_BY_CORE)
             filetype |= OFFLINE_FILE_TYPE_CORE_SHARDED;
+        if (encodings2regdeps_) {
+            filetype &= ~OFFLINE_FILE_TYPE_ARCH_ALL;
+            filetype |= OFFLINE_FILE_TYPE_ARCH_REGDEPS;
```

Comment:
This is in 2 different places now.  Let's make a filetype interface so any filter modifying the filetype will work, and remove this encodings2regdeps_.  Does that require a new API routine asking for the filetype from the filter?  That seems cleaner than this hardcoding for this one filter.

Reply:
Done. Added `add_to_filetype()` routine in `record_filter_func_t`. By default it doesn't change anything. The `encodings2regdeps_t` filter overrides it removing arch info and adding "arch regdeps" to the file_type.